homeassistant/components/mazda/__init__.py

```diff
@@ -0,0 +1,173 @@
+"""The Mazda Connected Services integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from pymazda import (
+    Client as MazdaAPI,
+    MazdaAccountLockedException,
+    MazdaAPIEncryptionException,
+    MazdaAuthenticationException,
+    MazdaException,
+    MazdaTokenExpiredException,
+)
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_REGION
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+from homeassistant.util.async_ import gather_with_concurrency
+
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Mazda Connected Services component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Mazda Connected Services from a config entry."""
+    email = entry.data[CONF_EMAIL]
+    password = entry.data[CONF_PASSWORD]
+    region = entry.data[CONF_REGION]
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+    mazda_client = MazdaAPI(email, password, region, websession)
+
+    try:
+        await mazda_client.validate_credentials()
+    except MazdaAuthenticationException:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH},
+                data=entry.data,
+            )
+        )
+        return False
+    except (
+        MazdaException,
+        MazdaAccountLockedException,
+        MazdaTokenExpiredException,
+        MazdaAPIEncryptionException,
+    ) as ex:
+        _LOGGER.error("Error occurred during Mazda login request: %s", ex)
+        raise ConfigEntryNotReady from ex
+
+    async def async_update_data():
+        """Fetch data from Mazda API."""
+
+        async def with_timeout(task):
+            async with async_timeout.timeout(10):
+                return await task
+
+        try:
+            vehicles = await with_timeout(mazda_client.get_vehicles())
+
+            vehicle_status_tasks = [
+                with_timeout(mazda_client.get_vehicle_status(vehicle["id"]))
+                for vehicle in vehicles
+            ]
+            statuses = await gather_with_concurrency(5, *vehicle_status_tasks)
+
+            for vehicle, status in zip(vehicles, statuses):
+                vehicle["status"] = status
+
+            return vehicles
+        except MazdaAuthenticationException as ex:
+            hass.async_create_task(
```

Comment:
Thinking now, we might want to just do a reload of the config entry here. Once we merge #45528, all entities will be marked automatically as `unavailable` while your `async_setup_entry` logic will trigger a re-auth and no longer try to make a request every 60 seconds.